fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SemiReorderRule.java

```diff
@@ -82,6 +87,26 @@ public boolean check(final OptExpression input, OptimizerContext context) {
             }
         }
 
+        // recompute new semi join and right child projection
+        OptExpression leftChildJoinRightChild = input.inputAt(0).inputAt(1);
+        ColumnRefSet leftChildJoinRightChildOutputColumns = leftChildJoinRightChild.getOutputColumns();
+
+        Projection leftChildJoinProjection = leftChildJoin.getProjection();
+        HashMap<ColumnRefOperator, ScalarOperator> rightExpression = new HashMap<>();
+        HashMap<ColumnRefOperator, ScalarOperator> semiExpression = new HashMap<>();
+        if (leftChildJoinProjection != null) {
+            for (Map.Entry<ColumnRefOperator, ScalarOperator> entry : leftChildJoinProjection.getColumnRefMap()
+                    .entrySet()) {
+                if (!entry.getValue().isColumnRef() &&
```

Comment:
`!entry.getValue().isColumnRef()`  is common

Reply:
done